_releases/2022-01-06-1.3.0-released.md

```diff
@@ -0,0 +1,208 @@
+require "syscall"
+
+module MySyscalls
+  Syscall.def_syscall write, Int32, fd : Int32, buf : UInt8*, count : LibC::SizeT
+end
+
+data = "Hello!\n"
+MySyscalls.write(1, data.to_unsafe, LibC::SizeT.new(data.size))
+{% endhighlight %}</div>
+
+## Multi-Assign
+
+Multiple assignments got improved in various ways. First, it's possible to use a splat target in a multi-assignment ([#10410](https://github.com/crystal-lang/crystal/pull/10410)):
+
+<div class="code_section">{% highlight crystal %}
+# Splat in multi-assign with array
+first, *rest, last = [1, 2, 3, 4, 5]
+first # => 1
+rest # => [2, 3, 4]
+last # => 5
+{% endhighlight %}</div>
+
+<div class="code_section">{% highlight crystal %}
+# Splat in multi-assign with tuple
+*rest, last = {"This", 15, 4, "tuple", true}
+rest # => {"This", 15, 4, "tuple"}
+last # => true
+{% endhighlight %}</div>
+
+To take the first and last element, it is possible to use the _underscore splat_ notation:
+<div class="code_section">{% highlight crystal %}
+# Ignoring the elements in the middle
+first, *_, last = {"This", 15, 4, "tuple", true}
+first # => "This"
+last # => true
+{% endhighlight %}</div>
+
+Second, there's an optional preview feature to detect unbalanced multi-assignments ([#11145](https://github.com/crystal-lang/crystal/pull/11145)). It can be enabled with the compiler flag `-Dstrict_multi_assign`.
+<div class="code_section">{% highlight crystal %}
+a, b, c = {1, 2} # Error: index out of bounds for Tuple(Int32, Int32) (2 not in -2..1)
+a, b = {1, 2, 3} # Error: cannot assign Tuple(Int32, Int32, Int32) to 2 targets
+{% endhighlight %}</div>
+
+Note that the first example fails with a similar error in 1.2.2 (this error is [expected to change](https://github.com/crystal-lang/crystal/issues/11695) in the near future).
+
+To ignore the extra items in strict mode you can use the underscore splat notation. This can be used to port existing code to work with the `strict_multi_assign` flag.
+<div class="code_section">{% highlight crystal %}
+a, b, *_ = {1, 2, 3} # Same as a, b = {1, 2, 3} without the flag
+{% endhighlight %}</div>
+
+⚠️ This second improvement is a breaking change; therefore it's not enabled by default. We encourage using it to detect possible errors in your code. This setting will likely be the default in 2.0. An additional restriction guarded by that flag is that if the right-hand side has a unique element, it must be of type `Indexable` ([#11545](https://github.com/crystal-lang/crystal/pull/11545)).
+
+## Number Autocast
+
+Primitive numeric values are now _autocasted_ to fit into larger types ([#11431](https://github.com/crystal-lang/crystal/pull/11431), [#11529](https://github.com/crystal-lang/crystal/pull/11529)). For instance, it is possible to call a function expecting an `Int64` with an `Int32` (note that before 1.3.0 only number _literals_ were autocasted):
+
+<div class="code_section">{% highlight crystal %}
+def foo(x : Int64)
+  x
+end
+
+foo 1_32 # Works in 1.2.2 and 1.3.0
```

Comment:
A little late, but shouldn't this be `1_i32`?

Reply:
🤦

Reply:
Technically `1_32` is still an Int32, so it's fine 😅